qiskit/primitives/base/base_result.py

```diff
@@ -45,43 +42,24 @@ def __post_init__(self) -> None:
             TypeError: If one of the data fields is not a Sequence or ``numpy.ndarray``.
             ValueError: Inconsistent number of experiments across data fields.
         """
+        num_experiments = None
         for value in self._field_values:  # type: Sequence
+            if num_experiments is None:
+                num_experiments = len(value)
             # TODO: enforce all data fields to be tuples instead of sequences
             if not isinstance(value, (Sequence, ndarray)) or isinstance(value, (str, bytes)):
                 raise TypeError(
                     f"Expected sequence or `numpy.ndarray`, provided {type(value)} instead."
                 )
-            if len(value) != self.num_experiments:
+            if len(value) != num_experiments:
                 raise ValueError("Inconsistent number of experiments across data fields.")
 
-    @property  # TODO: functools.cached_property when py37 is droppped
```

Comment:
In case someone stumbles over this (like I did): these are removed w/o deprecation warning since the deprecation PR went into `stable/0.46` directly -- as is the change from `BasePrimitiveResult` to `_BasePrimitiveResult`.

Reply:
yes, introduced by https://github.com/Qiskit/qiskit/pull/11054